### PR TITLE
Add error inputs to `gaussian_nll_loss` `OpInfo`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #88486

### CODE REVIEW

**Comment by nkaretnikov** [comment]
Split out the test changes from https://github.com/pytorch/pytorch/pull/85612 because other things are blocked.

**Review by lezcano — APPROVED** [review]
Cool!

File: test/test_nn.py [hunk 1]
```diff
@@ -6144,20 +6144,6 @@ def test_poisson_nll_loss_reduction_modes(self):
...
-        component_wise_loss = 0.5 * (torch.log(var) + (input - target)**2 / var)
-        self.assertEqual(component_wise_loss,
-                         F.gaussian_nll_loss(input, target, var, reduction='none'))
-        self.assertEqual(torch.sum(component_wise_loss),
-                         F.gaussian_nll_loss(input, target, var, reduction='sum'))
-        self.assertEqual(torch.mean(component_wise_loss),
-                         F.gaussian_nll_loss(input, target, var, reduction='mean'))
-        with self.assertRaisesRegex(ValueError, 'is not valid'):
-            F.gaussian_nll_loss(input, target, var, reduction='total')
-
```
Comment (nkaretnikov) [inline test/test_nn.py:6160]: Already handled by sample inputs.

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -6890,6 +6890,23 @@ def gen_shape_kwargs():
     for input, target, var, kwargs in gen_shape_kwargs():
         yield SampleInput(input, args=(target, var, ), kwargs=kwargs)
 
+def error_inputs_gaussian_nll_loss(op_info, device, **kwargs):
+    _make = partial(make_tensor, device=device, dtype=torch.float32)
```
Comment (lezcano) [inline torch/testing/_internal/common_methods_invocations.py:6894]: style nit. I think most opinfos call this `make_arg`, but not necessary to change it really.

**Pull request closed without merging**